gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/jupiter/core/v4/endpoint/loadbalancer/WeightDistributions.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@NoArgsConstructor
+@Getter
+@Setter
+public class WeightDistributions {
+
+    private int weightSum;
+    private int remainingSum;
+
+    private List<WeightDistribution> distributions = new ArrayList<>();
+
+    public WeightDistributions(final List<WeightDistribution> newDistributions) {
+        this.distributions = newDistributions;
+        int sum = 0;
+        for (WeightDistribution weightDistribution : this.distributions) {
+            sum += weightDistribution.getWeight();
+        }
+        weightSum = sum;
```

Comment:
Why not use a stream?

```suggestion
    weightSum = this.distributions.stream().mapToInt(WeightDistribution::getWeight).sum();
```

Reply:
I had issue with concurrent access when I first did that. Will try again.

Reply:
Even if foreach is old, and stream is "a la mode", this is not the best way to compute a sum. see following jmh result:

 - Result "io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer.WeightDistributionsBenchmark.forEach":
  24751,585 ±(99.9%) 382,631 ops/ms [Average]
  (min, avg, max) = (24649,898, 24751,585, 24862,176), stdev = 99,368
  CI (99.9%): [24368,954, 25134,217] (assumes normal distribution)

- Result "io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer.WeightDistributionsBenchmark.stream":
  15044,978 ±(99.9%) 190,343 ops/ms [Average]
  (min, avg, max) = (14973,409, 15044,978, 15102,103), stdev = 49,431
  CI (99.9%): [14854,636, 15235,321] (assumes normal distribution)